### DISCUSSION TITLE
Address Notes copy feature for filling arrays

### DISCUSSION THREAD

**Discussion by Keltron3030** [RetroAchievements/RAIntegration#781] [Ideas]
As a developer working on games with large memory arrays (objects, enemies, items), I would like to see a feature that would allow a developer to add repetitive notes to a large range en masse.  

For example, a game may have an array that spans 100s of addresses and it can be tedious to manually copy and paste mostly identical notes to all individual addresses.

I envision a `copy to range` button with two associated fields: 

1) A numerical field that will indicate how many addresses to skip between each note.
2) An address field that will indicate the last address to apply the note.

No start address field is needed, as the assumption would be that the currently selected and noted address is the starting address.  I would also like to see a numbering command that could be used in the notes in order to allow sequential numbering.  For example an initial array note called “Slot 1 enemy name (16-bit)” could be noted as “Slot &[seq] enemy name (16-bit)” which would positively increment by 1 for each note in the range.

Some considerations:

As a safe guard, the feature should only be allowed to write notes to an empty series.  If any address in the proposed series has a note, the write command should be aborted without any changes being made.  This would require the developer to clear any stray existing address notes to ensure they are only writing to a totally blank series and cannot unintentionally overwrite a large swath of notes.

The developer should receive a confirmation prompt prior to execution.

This feature should also be strictly developer only to avoid uninformed users from easily filling large amounts junk notes.

Simple MSPaint mockup
![1581093695815_meminsp](https://user-images.githubusercontent.com/42773860/74047766-dc6cce00-499e-11ea-9ebe-60b5a08d6306.png)

**Comment by televandalist** [comment]
So basically the memory inspector equivalent of dragging spreadsheet formulas. Sign me up.

**Comment by redwizard42** [comment]
I would, instead, prefer if I could apply an array note to a range of values, allowing me to specify the start, size, and bittage (is that a word? seems good), such that that one single note would appear on the code notes for any of the addresses listed. that way if I need to change the note(s), I'd only have to change it for the array as a whole, not for ever single byte.  Ideally, the code note (on site, for example), would expand to something like "Character HP Array [16-bit], Index 3, Low Byte" or something like that.

**Comment by Jamiras** [comment]
You can already apply a range to a code note, though it currently only impacts the search results (see #427). I plan to extend that to the highlighting in the memory view, but have no definite time frame for doing so.

But that is limited to a single size for the entire array. For the original example, the entire array might be 128 bytes, so putting the "[128-byte]" annotation onto the array would return individual bytes within the array as belonging to the array, but the offset would be number of bytes in, not number of 16-bit values in. Annotating it with "[16-bit] would only catch the first entry, and there's no way to combine both.
